src/lib.rs

```diff
@@ -1,11 +1,78 @@
 use cdshealpix as healpix;
 use cdshealpix::sph_geom::coo3d::{vec3_of, UnitVec3, UnitVect3};
+use moc::moc::range::CellSelection;
+use moc::moc::range::RangeMOC;
+use moc::qty::Hpx;
 use ndarray::{s, Array1, Zip};
-use numpy::{PyArrayDyn, PyArrayMethods};
+use numpy::{PyArray1, PyArrayDyn, PyArrayMethods, PyReadonlyArray1, PyReadonlyArray2};
 use pyo3::prelude::*;
 
 mod hierarchy;
 
+fn degrees_to_radians(degrees: f64) -> f64 {
+    degrees * (std::f64::consts::PI / 180.0)
+}
+
+#[pymodule]
+mod select {
```

Comment:
if I understand MOCs correctly (which I may not be!), these are only defined for "nested", so instead of a separate module we'd need to merge `select` into `nested`

Reply:
that makes sense. either that, or translate the cell ids to nested form

Reply:
Ok, so I'm trying to run your notebook, but I don't have geoarrow and am not able to easily install it? (pip install geoarrow-pyarrow geoarrow-pandas)

Reply:
you need `pip install 'geoarrow-rust-core>=0.4.0b3'` to reproduce (but `geoarrow-pyarrow` would also work, just with maybe a different API)

Reply:
ok I was able to run your notebook. If I change the polygon to

polygon_ = np.array([[153.56946903, -43.63459726],
       [153.56946902, -10.66818572],
       [113.33895308, -10.66818573],
       [113.33895309, -43.63459725],
       [153.56946904, -43.63459724]])
polygon_

then indeed it works. So this is the same bug as I encountered before.

Reply:
okay, thanks a lot. I'll have to check with the cdshealpix people on whether this is intentional